### ISSUE TITLE
How to implement LSQ using pytorch_quant

### ISSUE THREAD

**Issue by AnnaTrainingG** [NVIDIA/TensorRT-Model-Optimizer#54]
Hi, Are there any documents or recommendations available?
I've found the implementation of [LSQuantConv ](https://github.com/ZouJiu1/LSQplus/blob/master/quantization/lsqplus_quantize_V2.py)in PyTorch, but I'm unsure how to integrate it with quant_nn.QuantConv

```[tasklist]
### Tasks
```

**Comment by realAsma** [comment]
In ModelOpt quantization, the quantizer amax (`scale` is `amax/(maxvalue of the  quantized format)`) is a Pytorch buffer which is frozen during the training. 
To implement LSQ on your own, you would need to:
 
1. Update amax of [TensorQuantizer](https://nvidia.github.io/TensorRT-Model-Optimizer/reference/generated/modelopt.torch.quantization.nn.modules.tensor_quantizer.html#modelopt.torch.quantization.nn.modules.tensor_quantizer.TensorQuantizer) to be a Pytorch `nn.Parameter`.

2. Update the quantization functional you use such as [FakeTensorQuantFunction](https://nvidia.github.io/TensorRT-Model-Optimizer/reference/generated/modelopt.torch.quantization.tensor_quant.html#modelopt.torch.quantization.tensor_quant.LegacyFakeTensorQuantFunction) to return the gradient for amax as described in [LSQ](https://arxiv.org/pdf/1902.08153) paper.

**Comment by AnnaTrainingG** [comment]
ok thanks

**Comment by AnnaTrainingG** [comment]
Hello, I'm quite puzzled about the use of TensorRT (TRT) with INT8 quantization. Will the scale obtained from LSQ (Learned Step Size Quantization) actually be used in the real quantization process? Is the quantization process still performed within the FakeTensorQuantFunction? @realAsma

**Comment by realAsma** [comment]
> Hello, I'm quite puzzled about the use of TensorRT (TRT) with INT8 quantization. Will the scale obtained from LSQ (Learned Step Size Quantization) actually be used in the real quantization process? Is the quantization process still performed within the FakeTensorQuantFunction? [@realAsma](https://github.com/realAsma)

TRT export after INT8 quantization + LSQ would need some extra work. Currently the fake quantization functionals compute amax and pass that to the tensor quantization functional (see https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/ae915eaea9cb8a3f3438a68f2ff97f380b864f6e/modelopt/torch/quantization/tensor_quant.py#L330). 

To support LSQ, you would need to create a new quantization functional which takes the LSQ scale directly or have a method that converts the scale to amax so that LSQ is compatible with existing quantization functionals.

**Comment by realAsma** [comment]
@AnnaTrainingG I am closing this issue. I have added LSQ support to our feature roadmap. However this is deprioritized at the moment.
